iroh-base/src/key.rs

```diff
@@ -334,40 +334,19 @@ impl SecretKey {
         self.secret.verifying_key().into()
     }
 
-    /// Generate a new [`SecretKey`] with the default randomness generator.
-    pub fn generate() -> Self {
-        let mut rng = rand::rngs::OsRng;
-        Self::generate_with_rng(&mut rng)
-    }
-
     /// Generate a new [`SecretKey`] with a randomness generator.
-    pub fn generate_with_rng<R: CryptoRngCore + ?Sized>(csprng: &mut R) -> Self {
-        let secret = SigningKey::generate(csprng);
+    ///
+    /// ```rust
+    /// // use the OsRng option for OS depedndent most secure RNG.
+    /// let mut rng = rand::rngs::OsRng;
+    /// let _key = iroh_base::SecretKey::generate(&mut rng);
+    /// ```
+    pub fn generate<R: CryptoRngCore>(mut csprng: R) -> Self {
```

Comment:
The rand crate is on version 0.8.  Is it really a good idea to require it on the public API?

Reply:
all of the rust crypto world uses `rand` like this in their public interfaces.. I don't think there is much point in trying to do sth differently